paymentsheet/src/main/java/com/stripe/android/paymentsheet/PaymentOptionsActivity.kt

```diff
@@ -141,12 +147,59 @@ internal class PaymentOptionsActivity : BaseSheetActivity<PaymentOptionResult>()
             )
         }
 
-        continueButton.setOnClickListener {
-            viewModel.onUserSelection()
+        viewModel.primaryButtonOnPress.observe(this) { action ->
+            viewBinding.continueButton.setOnClickListener {
+                action()
+            }
+        }
+
+        viewModel.primaryButtonText.observe(this) { text ->
+            viewBinding.continueButton.setLabel(text)
         }
 
         viewModel.ctaEnabled.observe(this) { isEnabled ->
-            continueButton.isEnabled = isEnabled
+            viewBinding.continueButton.isEnabled = isEnabled
+        }
+
+        viewModel.ctaEnabled.observe(this) { isEnabled ->
+            viewBinding.continueButton.isEnabled = isEnabled
+        }
+
+        viewModel.updatePrimaryButtonText(
+            initial = true,
+            text = getString(R.string.stripe_paymentsheet_continue_button_label)
+        )
+
+        viewModel.updatePrimaryButtonOnPress(initial = true) {
+            viewModel.onUserSelection()
+        }
+    }
+
+    private fun setupNotes() {
+        viewModel.notesText.observe(this) { stringRes ->
+            val showNotes = stringRes != null
+            stringRes?.let {
+                viewBinding.notes.setContent {
+                    Html(
+                        html = getString(stringRes),
+                        imageGetter = mapOf(),
+                        color = PaymentsTheme.colors.subtitle,
+                        style = PaymentsTheme.typography.body1.copy(textAlign = TextAlign.Center)
+                    )
+                }
+            }
+            viewBinding.notes.isVisible = showNotes
```

Comment:
I think this might cause issues with our espresso tests. Can you try running them? They live here https://github.com/stripe/stripe-android/tree/master/paymentsheet-example/src/androidTestDebug/java/com/stripe/android

For context there's something about setting content to an invisible view that the idle resource manager doesn't like. I think you may have to sett the content after it is visible. but do check it might not be a problem.

Reply:
Ran these tests, seems ok :)